lib/utils/dialog/twake_dialog.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/widgets/twake_app.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:future_loading_dialog/future_loading_dialog.dart';
+
+import 'package:matrix/matrix.dart';
+
+class TwakeDialog {
+  static void hideLoadingDialog(BuildContext context) {
+    if (PlatformInfos.isWeb) {
+      TwakeApp.router.routerDelegate.pop();
+    } else {
+      Navigator.pop(context);
+    }
+  }
+
+  static void showLoadingDialog(BuildContext context) {
+    showGeneralDialog(
+      useRootNavigator: PlatformInfos.isWeb,
+      transitionDuration: const Duration(milliseconds: 700),
+      transitionBuilder: (context, animation, secondaryAnimation, child) {
+        return FadeTransition(
+          opacity: Tween<double>(begin: 0, end: 1).animate(animation),
+          child: WillPopScope(
+            onWillPop: () async => false,
+            child: const ProgressDialog(),
+          ),
+        );
+      },
+      context: context,
+      pageBuilder: (c, a1, a2) {
+        return const SizedBox();
+      },
+    );
+  }
+
+  static Future<LoadingDialogResult<T>> showFutureLoadingDialogFullScreen<T>({
+    required Future<T> Function() future,
+  }) async {
+    final twakeContext = TwakeApp.routerKey.currentContext;
+    if (twakeContext == null) {
+      Logs().e(
+        'TwakeLoadingDialog()::showFutureLoadingDialogFullScreen - Twake context is null',
+      );
+      return LoadingDialogResult<T>(
+        error: Exception('FutureDialog canceled'),
+        stackTrace: StackTrace.current,
+      );
+    }
+    return await showFutureLoadingDialog(
+      context: twakeContext,
+      future: future,
+    );
+  }
+
+  static Future<bool?> showDialogFullScreen({
+    required Widget Function() builder,
+  }) {
+    final twakeContext = TwakeApp.routerKey.currentContext;
+    if (twakeContext == null) {
+      Logs().e(
+        'TwakeLoadingDialog()::showDialogFullScreen - Twake context is null',
+      );
+    }
+    return showDialog(
+      context: twakeContext!,
+      builder: (context) => builder(),
+      barrierDismissible: true,
+      useRootNavigator: false,
+    );
+  }
+
+  static Future<bool?> showCupertinoDialogFullScreen({
+    required Widget Function() builder,
+  }) {
+    final twakeContext = TwakeApp.routerKey.currentContext;
+    if (twakeContext == null) {
+      Logs().e(
```

Comment:
the `showCupertinoDialog` still called if twakeContext is null

Reply:
done